src/main/java/com/redhat/devtools/intellij/knative/tree/KnFunctionsTreeStructure.java

```diff
@@ -230,7 +231,8 @@ private VirtualFile[] getModuleRoots(Module module) {
             roots.add(getModuleRootAsDirectory(contentRoot));
         }
         if (roots.isEmpty()) {
-            roots.add(getModuleRootAsDirectory(LocalFileSystem.getInstance().findFileByPath(new File(module.getModuleFilePath()).getParent())));
+            VirtualFile moduleDir = ProjectUtil.guessModuleDir(module);
+            roots.add(moduleDir);
         }
```

Comment:
@sbouchet, @jeffmaury, @lstocchi: I am very unsure about this replacement. Can you please enlighten me what this tried to do?

Reply:
By looking at the code it seems that it gets all modules belonging to the project, for each module it tries to get its module roots (`manager.getContentRoots`), if so it adds them to the roots array (which will be used to calculate which will be the roots in the plugin tree), if it doesn't find any root, we just get the parent of the moduleFile.

So by looking at the code in guessModuleDir -> https://github.com/JetBrains/intellij-community/blob/3814ce7f8db8f2e489bc8fcf0d60fd20475aec53/platform/ide-core/src/com/intellij/openapi/project/ProjectUtil.kt#L169-L172
I'm not sure if the value of `moduleFile?.parent` changes. 
I mean we are inside one project that could have multiple modules and we are looping through them. If they calculate moduleFile something like (get first module in the project) than it won't work. Bc it would always be the same.

Reply:
my similar fix in intellij-openshift was like it : 

![Capture d’écran du 2024-04-22 10-23-06](https://github.com/redhat-developer/intellij-knative/assets/600414/8fb8633f-4d6a-4b06-9b95-33294e4622a3)


https://github.com/redhat-developer/intellij-openshift-connector/pull/520/files

Reply:
keeping this untouched as discussed (see https://github.com/redhat-developer/intellij-knative/pull/255#pullrequestreview-2014106103)